NetworkRegistry/NetworkRegistry.md

```diff
@@ -0,0 +1,65 @@
+---
+CIP: TODO
+Title: Chain ID Registry
+Authors: Sebastien Guillemot <seba@dcspark.io>
+Comments-URI: TODO
+Status: Draft
+Type: Informational
+Created: 2021-11-24
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Currently Cardano has two easily-usable networks: "mainnet" and "testnet". However, in the future, we expect more networks to exist and so we need some way to refer to these networks to be able to write better multi-network applications and systems.
+
+## Terminology
+
+### Existing network IDs
+
+Cardano currently has three ways to refer to a network:
+- The "network ID" included in every address and also optionally present in the transaction body. This only stores 16 possibilities (4 bits)
+- The "network magic" used for Byron addresses and in the handshake with other nodes in the network layer. This is a random 32-bit number
+- The genesis block hash (a 28-byte number)
+
+## Motivation
+
+Blockchains can have multiple deployments of the same codebase. For example:
+
+1. Test networks where the base asset has no value (so devs can test at no cost)
+1. Test networks where the protocol is simplified for ease of testing (ex: Cardano but with a Proof of Authority consensus for stable block production in testing)
+1. Test networks for new features
+    1. Plutus Application Backend (PAB) testnet
+    2. Shelley incentivized testnet
+1. Forks that diverge in feature set (the many forks of Bitcoin and Ethereum)
+
+dApps may be deployed on specific testnets that match their criteria and wallets need to know about these networks to know how to behave.
+
+Additionally, having a standardized registry for networks allows easy integration into the broader crypto ecosystem via standards like [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md)
+
+## Specification
+
+We create a machine-readable registry of networks
```

Comment:
Where will this be stored? How will the data be governed? Will it be decentralized?

Or is this proposal just about the format of the data, leaving the actual form of the _registry_ unspecified?

Reply:
It will be stored here in this CIP. The governance is that anybody who submits a chain ID for any actual network they launched gets it. Similar to other blockchains that have similar IDs with similar governance. This system will not be uniquely associating chain IDs to projects. It's just a way for people to declare they are using an ID in an agreed-upon place to avoid people accidentally colliding.
This is similar to the way other chains deal with chain IDs and similar to other registries like the metadata label registry (CIP10) and SLIP44.

Reply:
Ah, that makes sense. It would be good to have something about t hat in the text!

What about short-lived networks, e.g. networks spun up for testing and then spun down shortly afterwards? Is the idea that they should use the same format to identify themselves, but not bother to register here? Will that affect how e.g. wallets interact with such networks?

Reply:
Yeah I guess short-lived networks could just pick a IDs to avoid any chance of collision and they can double-check this CIP to make sure they're not conflicting. We could reserve one of the network IDs for short-term networks and then have them pick a random network magic within it if that sounds like something useful